File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    for s, e, k in queries:
        tmp = []
        for i in range(s, e + 1):
            if arr[i] > k:
                tmp.append(arr[i])
        answer.append(min(tmp) if tmp else -1)
    return answer

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    vector<int> answer;
    
    for (auto q : queries) {
        vector<int> tmp;
        for (int i = q[0]; i <= q[1]; i++)
            if (arr[i] > q[2]) tmp.push_back(arr[i]);
        answer.push_back(tmp.size() > 0 ? *min_element(tmp.begin(), tmp.end()) : -1);
    }
    
    return answer;
}